account_mass_reconcile/models/base_reconciliation.py

```diff
@@ -0,0 +1,182 @@
+# © 2012-2016 Camptocamp SA (Guewen Baconnier, Damien Crier, Matthieu Dietrich)
+# © 2010 Sébastien Beau
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from operator import itemgetter
+from functools import reduce
+
+from odoo import models, api, fields
+from odoo.tools.safe_eval import safe_eval
+
+
+class MassReconcileBase(models.AbstractModel):
+    """Abstract Model for reconciliation methods"""
+    _name = 'mass.reconcile.base'
+    _inherit = 'mass.reconcile.options'
+
+    account_id = fields.Many2one(
+        'account.account',
+        string='Account',
+        required=True
+    )
+    partner_ids = fields.Many2many(
+        comodel_name='res.partner',
+        string='Restrict on partners',
+    )
+    # other fields are inherited from mass.reconcile.options
+
+    @api.multi
+    def automatic_reconcile(self):
+        """ Reconciliation method called from the view.
+
+        :return: list of reconciled ids
+        """
+        self.ensure_one()
+        return self._action_rec()
+
+    def _action_rec(self):
+        """ Must be inherited to implement the reconciliation
+
+        :return: list of reconciled ids
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def _base_columns():
+        """ Mandatory columns for move lines queries
+        An extra column aliased as ``key`` should be defined
+        in each query."""
+        aml_cols = (
+            'id',
+            'debit',
+            'credit',
+            'date',
+            'ref',
+            'name',
+            'partner_id',
+            'account_id',
+            'reconciled',
+            'move_id')
+        return ["account_move_line.{}".format(col) for col in aml_cols]
+
+    def _select(self, *args, **kwargs):
```

Comment:
wouldn't be better to prefix these methods' names w/ something to avoid possible clashes w/ odoo internals?